examples/groups/README.md

```diff
@@ -3,6 +3,56 @@
 This example creates policies and various different groups.
 
 
+## Overview
+
+You can define as many groups as desired and reference them by their names in **[`var.users`](../users/)** in order to attach as many groups to a specific user as needed.
+* When using the `policies` key, respective policies must be defined in **[`var.policies`](../policies/)**.
+
+
+## Examples
+
+**Note:** The following examples only shows the creation of a single group.
+You can however create as many groups as desired. Also re-arranging them within the list will not
+trigger terraform to change or destroy resources as they're internally stored in a map (rather than a list) by their group names as keys (See module's `locals.tf` for transformation).
+
+Groups are defined as follows:
+
+`terraform.tfvars`
+```hcl
+groups = [
+  {
+    name       = "group-name" # Name of the group (reference this in var.users for attachment)
+    path       = "/path/"     # Defaults to 'var.group_path' if variable is set to null
+    policies   = [
+      "policy-name-1",        # policy-name-1 must be defined in var.policies
+      "policy-name-2",        # policy-name-2 must be defined in var.policies
+    ]
+    policy_arns = [           # Attach policies by ARN
+      "arn:aws:iam::aws:policy/AmazonEC2FullAccess",
+      "arn:aws:iam::aws:policy/AWSResourceAccessManagerFullAccess",
+    ]
+    inline_policies = [       # Attach inline policies defined via JSON files
+      {
+        name = "inline-policy-1"
+        file = "data/policies/kms-ro.json"
+        vars = {}
+      },
+      {
+        name = "inline-policy-2"
+        file = "data/policies/sqs-ro.json.tmpl"
+        vars = {  # You can use variables inside JSON files
```

Comment:
Nicely done but it kind of eats up the unique selling point of generating the policy using the data source instead :)

Reply:
@danvaida This is the same behaviour as before.

**But**, I've included an example that shows how that will be possible nonetheless (additionally to the above specification):

https://github.com/cytopia/terraform-aws-iam/blob/master/examples/policies-with-custom-data-sources/main.tf